src/shared_types.d.ts

```diff
@@ -0,0 +1,121 @@
+type NodeRef = MQNode | 0;
```

Comment:
I know that this typescript conversion was trying to preserve most of the structure of the original codebase, but handling of null/undefined I think is something I'd like to discuss going forward. I don't know why 0 was chosen as a special value originally instead of just using undefined or null when a NodeRef isn't set.

I've been trying to see what the typescript pattern might exist that would accomplish something like Scala's Option type. It's not a perfect solution to null pointers, but it does give an explicit compile hint that accessors of a property/object should be handling a potential missing value.

https://www.scala-lang.org/api/2.13.6/scala/Option.html

Reply:
I think 0 was chosen originally because you can attempt to access properties on a number without throwing an error. For instance: https://github.com/desmosinc/mathquill/blob/9de1943220f8d0dcf401ac27e751e9544b4a0c5c/src/services/keystroke.js#L268

This won't throw an error if `cursor[L] === 0` but would throw an error if `cursor[L] === undefined`.

Reply:
Tracked as https://github.com/mathquill/mathquill/issues/981